### PR TITLE
Linter and Prettier

### PR DESCRIPTION
**Pull request by lananelson**
setup commands and pre-commit hooks for enforcing linting and formatting rules

Once the rules were set up I ran
```
npm run lint
npm run format
```

which formatted a bunch of files according to the rules

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -1,3 +1,13 @@
 {
-  "extends": "next/core-web-vitals"
+  "settings": {
+    "react": {
+      "version": "detect"
+    }
+  },
+  "extends": ["next/core-web-vitals", "eslint:recommended", "plugin:@typescript-eslint/recommended", "prettier"],
```
Comment (lananelson) [inline .eslintrc.json:7]: @JoHeyJo most rules come from this little line. E.g. `"plugin:@typescript-eslint/recommended"` results in these rules applied across our project https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/src/configs/eslint-recommended.ts

Comment (JoHeyJo) [reply .eslintrc.json:7]: @lananelson Cool! So we're using the recommended config. Is the code below custom rules or just more config code?

**Comment by JoHeyJo** [comment]
@lananelson LGTM!

File: .eslintrc.json (hunk 1 above)
Comment (lananelson) [reply .eslintrc.json:7]: the code below has no rules yet `rules: {}` but it does help to set up typescript parser options

**Pull request merged**